components/Events/Filters.js

```diff
@@ -0,0 +1,228 @@
+  });
+  const [chosenDays, setChosenDays] = useState({
+    Sunday: false,
+    Monday: false,
+    Tuesday: false,
+    Wednesday: false,
+    Thursday: false,
+    Friday: false,
+    Saturday: false,
+  });
+  const [chosenTimes, setChosenTimes] = useState({
+    fromTime: '',
+    toTime: '',
+  });
+  const [chosenLoc, setChosenLoc] = useState({
+    'Online': false,
+    'In-Person': false,
+  });
+
+  const dateToMinutes = (date) => {
+    const itemDate = new Date(date);
+    const itemMins =
+      itemDate.getHours() * 60 + itemDate.getMinutes();
+    return itemMins;
+  };
+
+  const handleSearch = () => {
+    // Check search input
+    const lowerCaseSearch = searchValue.toLowerCase();
+    let allEvents = events.filter((item) =>
+      [item.title, item.committee, item.location, item.description].some(
+        (itemsToSearch) =>
+          String(itemsToSearch).toLowerCase().includes(lowerCaseSearch),
+      ),
+    );
+    // Check time filter
+    const isFilterTimeEmpty = Object.keys(chosenTimes).every(
+      (time) => chosenTimes[time] === '',
+    );
+    if (!isFilterTimeEmpty) {
+      const filterFromMins =
+        parseInt(chosenTimes.fromTime.split(':')[0]) * 60 +
+        parseInt(chosenTimes.fromTime.split(':')[1]);
+      const filterToMins =
+        parseInt(chosenTimes.toTime.split(':')[0]) * 60 +
+        parseInt(chosenTimes.toTime.split(':')[1]);
+
+      allEvents = allEvents.filter((item) => {
+        const itemStartMins = dateToMinutes(item.start);
+        const itemEndMins = dateToMinutes(item.end);
+        return itemStartMins >= filterFromMins && itemEndMins <= filterToMins;
+      });
+    }
+    // Check committee filter
+    const isCommEmpty = Object.keys(chosenComms).every(
+      (comm) => !chosenComms[comm],
+    );
+    if (!isCommEmpty) {
+      allEvents = allEvents.filter((item) => {
+        if (!item.committee) return false;
+        return chosenComms[item.committee];
+      });
+    }
+    // Check days filter
+    const isDaysEmpty = Object.keys(chosenDays).every(
+      (day) => !chosenDays[day],
+    );
+    if (!isDaysEmpty) {
+      allEvents = allEvents.filter((item) => {
+        const date = new Date(item.start);
+        const itemDay = Object.keys(chosenDays)[date.getDay()];
+        return chosenDays[itemDay];
+      });
+    }
+    // Check location filter
+    const isLocEmpty = Object.keys(chosenLoc).every((loc) => !chosenLoc[loc]);
+    if (!isLocEmpty) {
+      allEvents = allEvents.filter((item) => {
+        const itemLoc = item.location.toLowerCase().includes('zoom') ? 'Online' : 'In-Person';
+        return chosenLoc[itemLoc];
+      });
+    }
+
+    const newEvents = allEvents.map((original_event, index) => ({
+      ...original_event,
+      id: index,
+    }));
+    props.handleChange(newEvents);
+  };
+
+  const handleWeekdayChange = (day) => {
+    const updated = { ...chosenDays };
+    updated[day] = !updated[day];
+    setChosenDays(updated);
```

Comment:
you can do something like this! can apply to other parts also
```suggestion
    setChosenDays({ ...chosenDays, [day]: !chosenDays[day] });
```